alicloud/resource_alicloud_ots_table.go

```diff
@@ -0,0 +1,157 @@
+package alicloud
+
+import (
+	"fmt"
+	"github.com/aliyun/aliyun-tablestore-go-sdk/tablestore"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"time"
+)
+
+func resourceAlicloudOtsTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunOtsTableCreate,
+		Read:   resourceAliyunOtsTableRead,
+		Update: resourceAliyunOtsTableUpdate,
+		Delete: resourceAliyunOtsTableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"table_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"primary_key": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validatePrimaryTypeKey,
```

Comment:
I think validatePrimaryTypeKey can be replaced by unified method validateAllowedStringValue.